### PR TITLE
[TypeDeclaration] Add PropertyTypeFromStrictSetterGetterRector

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: rules/TypeDeclaration/Rector/Class_/PropertyTypeFromStrictSetterGetterRector.php [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            }
+
+            $propertyTypeDclaration = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode(
+                $getterSetterPropertyType,
+                TypeKind::PROPERTY
+            );
+            if (! $propertyTypeDclaration instanceof Node) {
+                continue;
+            }
+
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/Class_/PropertyTypeFromStrictSetterGetterRector.php:113]: Make property type guard is needed here

https://github.com/rectorphp/rector-src/blob/b98de3edc21312c2e5483b6a120ccb04642f5346/rules/TypeDeclaration/Rector/Property/TypedPropertyFromAssignsRector.php#L110

Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/Class_/PropertyTypeFromStrictSetterGetterRector.php:113]: I see. Could you handle this and add a test case too?

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/Class_/PropertyTypeFromStrictSetterGetterRector.php:113]: I will try.

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/Class_/PropertyTypeFromStrictSetterGetterRector.php:113]: here the PR https://github.com/rectorphp/rector-src/pull/3319

**Pull request merged**